create_language_model.py

```diff
@@ -0,0 +1,241 @@
+	if counter != NULL:
+	    counter += 1
+	if counter > 1:
+	    chars = line.split()[1:]
+	    for char in chars:
+		if char == 'XX':
+		    charsmap_table['XX' + str(xx_counter)] = CTRL # The 'XX' in the charset table indicates a nonexisting character
+		    xx_counter += 1
+		elif char == 'lf':
+		    charsmap_table['\r'] = CRLF
+		elif char == 'cr':
+		    charsmap_table['\n'] = CRLF
+		elif char == 'sp':
+		    charsmap_table[' '] = SYMBOL
+		elif len(char) == 1:
+		    if char.isdigit():
+			charsmap_table[char] = DIGIT
+		    elif char.isalpha():
+			if not keep_english_letter and char in english_letters:
+			    charsmap_table[char] = CTRL
+			else:
+			    charsmap_table[char] = ALPHA
+		    else:
+			charsmap_table[char] = SYMBOL
+		else:
+		    charsmap_table[char] = CTRL
+    f.close()
+    
+if counter == NULL:
+    print "Wrong/unknown charset name! Please check the 'CharsetsTabs.txt' file."
+    sys.exit(1)
+
+# *************************************************************** CHARS FREQUENCY TABLE *************************************************************** #
+letters_table = []
+for char in charsmap_table:
+    if charsmap_table[char] == ALPHA:
+	letters_table.append(char)
+
+chars_frequency_table = OrderedDict([])
+for char in charsmap_table: 
+    chars_frequency_table[char] = 0
+
+lang_name = os.path.splitext(os.path.basename(file_name))[0] 
+print "Creating chars frequency table for %s language" % lang_name
+with codecs.open(file_name, 'r', encoding='utf-8') as f:
+    for line in f:
+	for char in line:
+	    if char in chars_frequency_table:
+		chars_frequency_table[char] += 1
+    f.close()
+
+counter = 0
+frequent_letters_table = []
+for char, chars_count in sorted(chars_frequency_table.iteritems(), key=operator.itemgetter(1), reverse=True):
+    if char in letters_table:
+	counter += 1
+	charsmap_table[char] = counter
+	if counter <= SAMPLE_SIZE:
+	    frequent_letters_table.append(char)
+
+# ****************************************************************** CHARMAP OUTPUT ******************************************************************* #
+with open("lang" + lang_name + "model.py", 'w') as f:
+    f.write("%s_%s_char_to_order_map = (\n" % (normalize_charset_name(charset_name), first_letter_to_uppercase(lang_name)))
+    f.write("  #0   1   2   3   4   5   6   7   8   9   A   B   C   D   E   F\n  ")
+    counter = 1
+    for item in charsmap_table:
+	f.write("%3d," % charsmap_table[item])
+	if counter % 16 == 0:
+	    f.write("  # %2X\n  " % int(counter-16))
+	counter += 1
+    f.write(")\n\n")
+    f.close()
+
+# *************************************************************** BIGRAM FREQUENCY TABLE ************************************************************** #
+twochars_sequences_frequency_table = OrderedDict([])
+for char1 in frequent_letters_table:
+    for char2 in frequent_letters_table:
+	twochars_sequences_frequency_table[char1+char2] = 0
+
+print "Creating bigrams frequency table for %s language" % lang_name
```

Comment:
chardet needs to maintain cross-compatibility with Python 2 and 3, and in Python 3, `print` is a function, so please change these to:

```python
print("Creating bigrams frequency table for %s language" % lang_name)
```